app/src/main/java/utils/QueryUtils.java

```diff
@@ -150,13 +162,65 @@ public void done(List<Post> posts, ParseException e) {
                 }
                 for (Post post : posts) {
                     Log.i(TAG, "Content: " + post.getKeyUser());
+                    post.getKeyUser().pinInBackground();
                 }
+                AsyncTask.execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        Log.i(TAG, "Saving data into the database");
+                        List<RoomPost> roomPosts = new ArrayList<>();
+                        for (Post post : posts) {
+                            RoomPost roomPost = new RoomPost(post);
+                            roomPosts.add(roomPost);
+                        }
+                        List<RoomUser> roomUsers = RoomPostWithObjects.usersFromRoomPosts(roomPosts);
+                        // Must load in users before posts in order for foreign key to work
+                        ROOM_POST_DAO.insertModel(roomUsers.toArray(new RoomUser[0]));
+                        ROOM_POST_DAO.insertModel(roomPosts.toArray(new RoomPost[0]));
+                    }
+                });
                 allPosts.addAll(posts);
                 adapter.notifyDataSetChanged();
             }
         });
     }
 
+    public static void queryLongPostsOffline(Context context, List<BasePost> allPosts,
+                                             PostAdapter adapter, Group currentGroup) {
+        adapter.clear();
```

Comment:
should we clear the screen first, then do some processing to fetch items from the db before resurfacing everything? there would be some kind of flicker here that may not be desirable. i think you also never clear allposts so technically your list would still contain previous items.

maybe instead, let allPosts remain the source of truth. you should clear the list and not the screen quite yet. add the necessary posts from your db, then notify the adapter of changes.